slurk/models/user.py

```diff
@@ -50,98 +50,103 @@ def join_room(self, room):
         if self.session_id is not None:
             join_room(str(room.id), self.session_id, "/")
 
-            socketio.emit(
-                "joined_room",
-                {
-                    "room": str(room.id),
-                    "user": self.id,
-                },
-                room=self.session_id,
-            )
-
-            socketio.emit(
-                "status",
-                dict(
-                    type="join",
-                    user=dict(id=self.id, name=self.name),
+            def joined():
+                current_app.session.add(self)
+                current_app.session.add(room)
+
+                socketio.emit(
+                    "status",
+                    dict(
+                        type="join",
+                        user=dict(id=self.id, name=self.name),
+                        room=str(room.id),
+                        timestamp=str(datetime.utcnow()),
+                    ),
                     room=str(room.id),
```

Comment:
```suggestion
            def joined(user, room_id):
                socketio.emit(
                    "status",
                    dict(
                        type="join",
                        user=dict(id=user["id"], name=user["name"]),
                        room=room_id,
                        timestamp=str(datetime.utcnow()),
                    ),
                    room=str(room_id),
```

Reply:
I tried to avoid passing back the user to the server as you could change the JavaScript that they pass any user back, which then is passed to any other client with `emit("status")`. We would need to verify this, so another call to the database would have to be made.